lib/selective/ruby/core/controller.rb

```diff
@@ -153,6 +152,31 @@ def spawn_transport_process(url)
           end
         end
 
+        def wait_for_connectivity
+          @connectivity = false
+
+          Thread.new do
+            sleep(5)
+            return if @connectivity
+
+            puts "Transport process failed to start. Exiting..."
+            kill_transport
+            exit
+          end
+
+          loop do
+            message = pipe.read
+
+            # The message is nil until the transport opens the pipe
+            # for writing. So, we must handle that here.
+            next sleep(0.1) if message.nil?
+            
+            response = JSON.parse(message, symbolize_names: true)
+            @connectivity = true if response[:command] == "connected"
```

Comment:
This is to be paired with a change in the backend that sends the "connected" command as soon as the socket connects.